### ISSUE TITLE
Suggestion for generated C++ code

### ISSUE THREAD

**Issue by stefanos82** [natalie-lang/natalie#1055]
Hey @seven1m, I hope you are doing well.

I have started watching your videos and decided to play with Natalie a bit; great work by the way, I'm learning a lot around this topic.

I have noticed in generated C++ code that you silence some variables with `(void)` to make the compiler stop throwing warnings.

Since you are using C++17, why not declare your variable(s) that could throw warnings with `[[maybe_unused]]`?

For instance, lines such as

```c++
Value self = GlobalEnv::the()->main_obj();
    (void)self; // don't warn about unused var
```

could be set as

```c++
[[maybe_unused]] Value self = GlobalEnv::the()->main_obj();
```

What do you think?

**Comment by seven1m** [comment]
Yeah that's a good idea. If you want to give it a try, please go for it!

**Comment by stefanos82** [comment]
I know absolutely **nothing** about Ruby syntax :laughing: I've just discovered your channel via @awesomekling and thought it would be cool to check your work.

Is there any path I should follow to fully understand Ruby's logic, so I can start getting a rough idea how it actually works behind the scenes as a dynamic language?

**Comment by stefanos82** [comment]
By the way @seven1m, I opened a PR about it in case you want to give it a try.

**Comment by seven1m** [comment]
Fixed in be802d63fb16dbd311264f0974c6375802d53aa2. Thanks!
